base/applications/screensavers/logon/logon.c

```diff
@@ -18,127 +18,93 @@
  *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
-#include <stdarg.h>
+#include <stdlib.h>
 #include <windef.h>
 #include <winbase.h>
 #include <wingdi.h>
 #include <winuser.h>
 #include <scrnsave.h>
-#include <stdlib.h>
-#include <tchar.h>
 
 #include "resource.h"
 
-#define RANDOM( min, max ) ((rand() % (int)(((max)+1) - (min))) + (min))
+#define RANDOM(min, max) ((rand() % (int)(((max)+1) - (min))) + (min))
 
-#define APPNAME               _T("Logon")
 #define APP_TIMER             1
-#define APP_TIMER_INTERVAL    2000
+#define APP_TIMER_INTERVAL    10000
 
-HBITMAP
+static HBITMAP
 GetScreenSaverBitmap(VOID)
 {
-    OSVERSIONINFOEX osvi;
-
-    ZeroMemory(&osvi, sizeof(OSVERSIONINFOEX));
+    OSVERSIONINFOEX osvi = {0};
     osvi.dwOSVersionInfoSize = sizeof(OSVERSIONINFOEX);
     GetVersionEx ((OSVERSIONINFO *) &osvi);
 
-    switch(osvi.wProductType)
-    {
-        case VER_NT_WORKSTATION:
-            return LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(IDB_WORKSTATION), IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
-            break;
-        default:
-            return LoadImage(GetModuleHandle(NULL), MAKEINTRESOURCE(IDB_SERVER), IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
-            break;
-    }
+    return LoadImageW(GetModuleHandle(NULL),
+                      MAKEINTRESOURCEW(osvi.wProductType == VER_NT_WORKSTATION ?
+                                       IDB_WORKSTATION : IDB_SERVER),
```

Comment:
Prefer doing the logic thing outside of the macro:
```suggestion
                      osvi.wProductType == VER_NT_WORKSTATION ?
                          MAKEINTRESOURCEW(IDB_WORKSTATION)   :
                          MAKEINTRESOURCEW(IDB_SERVER),
```
(or something similar, no formatting preference here)